src/declarativewebcontainer.cpp

```diff
@@ -139,6 +135,10 @@ void DeclarativeWebContainer::setWebPage(DeclarativeWebPage *webPage)
         m_webPage = webPage;
 
         if (m_webPage) {
+            connect(m_webPage, SIGNAL(navigationHistoryChanged()), this, SIGNAL(canGoForwardChanged()));
+            connect(m_webPage, SIGNAL(navigationHistoryChanged()), this, SIGNAL(canGoBackChanged()));
```

Comment:
At some point of time would be good to split navigationHistoryChanged into two signals like canGoForwardChanged and canGoBackChanged (qtmozembed).